From 17b4d87853f9f20c2bc124da6594f760b4f2a927 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Tue, 27 Jun 2023 10:57:32 +0530
Subject: [PATCH] Fix for Local Media Player in Passenger Zone.

Currently MultiDisplay launcher does not have capability
to show launcher icons for media services app, as these apps
does not have any launcher category, So it does not show icon in
launcher app.

Querying all MediaBrowsingService apps and adding them in launcher.

Tracked-On: OAM-106796
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 .../common/source/MediaSourceViewModel.java      | 16 ++++++++++++----
 .../common/source/MediaSourceViewModelTest.java  |  3 ++-
 2 files changed, 14 insertions(+), 5 deletions(-)

diff --git a/car-media-common/src/com/android/car/media/common/source/MediaSourceViewModel.java b/car-media-common/src/com/android/car/media/common/source/MediaSourceViewModel.java
index 0b297fea6..00fb22b57 100644
--- a/car-media-common/src/com/android/car/media/common/source/MediaSourceViewModel.java
+++ b/car-media-common/src/com/android/car/media/common/source/MediaSourceViewModel.java
@@ -20,6 +20,7 @@ import static com.android.car.apps.common.util.LiveDataFunctions.dataOf;
 
 import android.app.Application;
 import android.car.Car;
+import android.car.Car.CarServiceLifecycleListener;
 import android.car.CarNotConnectedException;
 import android.car.media.CarMediaManager;
 import android.content.ComponentName;
@@ -64,7 +65,7 @@ public class MediaSourceViewModel extends AndroidViewModel {
         MediaBrowserConnector createMediaBrowserConnector(@NonNull Application application,
                 @NonNull MediaBrowserConnector.Callback connectedBrowserCallback);
 
-        Car getCarApi();
+        Car getCarApi(CarServiceLifecycleListener mCarServiceLifecycleListener);
 
         CarMediaManager getCarMediaManager(Car carApi) throws CarNotConnectedException;
 
@@ -94,8 +95,9 @@ public class MediaSourceViewModel extends AndroidViewModel {
             }
 
             @Override
-            public Car getCarApi() {
-                return Car.createCar(application);
+            public Car getCarApi(CarServiceLifecycleListener mCarServiceLifecycleListener) {
+                return Car.createCar(application, /* handler= */ null,
+                Car.CAR_WAIT_TIMEOUT_WAIT_FOREVER, mCarServiceLifecycleListener);
             }
 
             @Override
@@ -123,7 +125,13 @@ public class MediaSourceViewModel extends AndroidViewModel {
 
         mMode = mode;
         mInputFactory = inputFactory;
-        mCar = inputFactory.getCarApi();
+        mCar = inputFactory.getCarApi((car, ready) -> {
+            if (!ready) {
+                Log.d(TAG, "Disconnected from Car Service");
+                return;
+            }
+            Log.d(TAG, "Connected to Car Service");
+        });
 
         mBrowserConnector = inputFactory.createMediaBrowserConnector(application, mBrowserCallback);
 
diff --git a/car-media-common/tests/robotests/src/com/android/car/media/common/source/MediaSourceViewModelTest.java b/car-media-common/tests/robotests/src/com/android/car/media/common/source/MediaSourceViewModelTest.java
index 196553fcd..3403a0602 100644
--- a/car-media-common/tests/robotests/src/com/android/car/media/common/source/MediaSourceViewModelTest.java
+++ b/car-media-common/tests/robotests/src/com/android/car/media/common/source/MediaSourceViewModelTest.java
@@ -28,6 +28,7 @@ import static org.robolectric.RuntimeEnvironment.application;
 
 import android.app.Application;
 import android.car.Car;
+import android.car.Car.CarServiceLifecycleListener;
 import android.car.media.CarMediaManager;
 import android.content.ComponentName;
 import android.support.v4.media.MediaBrowserCompat;
@@ -102,7 +103,7 @@ public class MediaSourceViewModelTest {
             }
 
             @Override
-            public Car getCarApi() {
+            public Car getCarApi(CarServiceLifecycleListener mCarServiceLifecycleListener) {
                 return mCar;
             }
 
-- 
2.17.1

